### PR TITLE
Part of Issue #303: Improve contract endpoint error handling

### PR DESCRIPTION
**Pull request by jahabeebs**
part of this PR: https://github.com/UseKeyp/keyp-app/pull/306

## Description
- Edit TS types to add optional error field, make certain fields mandatory that were nullable before
- Add try/catch similar to /tokens helpers

## Screenshots

Transfer error (no helpful error message yet because testing against prod)

transfer error

Test write contract success

test write contract

### CODE REVIEW

File: packages/js-sdk/src/contract-helpers.ts [hunk 1]
```diff
@@ -62,6 +65,9 @@ const readContract = async ({accessToken, address, abi, args}: ReadContractParam
             explorerUrl: response.data.explorerUrl,
             response: response.data.response,
         };
+    } catch (error) {
+        return { status: 'FAILURE', explorerUrl: '', response: { type: '', hex: '' }, error: error };
+    }
```
Comment (pi0neerpat) [inline packages/js-sdk/src/contract-helpers.ts:70]: So is the purpose of this to ensure that the return type is always the same, even if the API server doesnt return the expected types?

**Review by pi0neerpat — APPROVED** [review]
Nice!

File: packages/js-sdk/src/contract-helpers.ts (hunk 1 above)
Comment (jahabeebs) [reply packages/js-sdk/src/contract-helpers.ts:70]: right, these values will be an empty string when there's a failure by the API anyway but for clarity we can be explicit about it so the user looking at the function knows what to handle beyond just looking at the TS types

Comment (pi0neerpat) [reply packages/js-sdk/src/contract-helpers.ts:70]: Awesome thanks!

**Pull request merged**